examples/search_bar/main.rs

```diff
@@ -0,0 +1,62 @@
+use gtk::prelude::*;
+use gtk::glib::clone;
+
+fn main() {
+    let application = gtk::Application::new(
+        Some("com.github.gtk-rs.examples.search_bar"),
+        Default::default(),
+    );
+    application.connect_activate(build_ui);
+    application.run();
+}
+
+fn build_ui(application: &gtk::Application) {
+    let window = gtk::ApplicationWindow::new(application);
+    window.set_default_size(400, 400);
+    window.set_title(Some("Search Bar"));
+
+    let header_bar = gtk::HeaderBar::new();
+    window.set_titlebar(Some(&header_bar));
+
+    let search_button = gtk::ToggleButton::new();
+    search_button.set_icon_name("system-search-symbolic");
+    header_bar.pack_end(&search_button);
+
+    let window_box = gtk::Box::new(gtk::Orientation::Vertical, 6);
+    window.set_child(Some(&window_box));
+
+    let search_bar = gtk::SearchBar::new();
+    window_box.append(&search_bar);
+    search_bar.set_valign(gtk::Align::Start);
+    search_bar.set_key_capture_widget(Some(&window));
+
+    let entry = gtk::SearchEntry::new();
+    entry.set_hexpand(true);
+    search_bar.set_child(Some(&entry));
+
+    let label = gtk::Label::new(Some("Type to start search"));
+    label.set_hexpand(true);
+    window_box.append(&label);
+
+    search_button.connect_toggled(clone!(@weak search_bar => move |_| {
```

Comment:
This can be probably use [bind_property](https://gtk-rs.org/gtk-rs/git/docs/glib/object/trait.ObjectExt.html#tymethod.bind_property) API. Something like `search_button.bind_property("active", &search_bar, "search-mode-enabled").build();` should work I guess.